### PR TITLE
diagnostic plot for normal likelihood and normal prior mixture

### PR DESCRIPTION
**Pull request by LSun**
diag.ash(ash_obj) produces a Q-Q plot for estimated Fhat vs uniform, which should look like a straight line from 0 to 1 under goodness of fit.
please check it makes sense.
can add more cosmetic effect to the plot later.

### CODE REVIEW

File: R/ashutility.R [hunk 1]
```diff
@@ -57,6 +57,28 @@ plot.ash = function(x,...,xmin,xmax){
...
+  cdfhat = cdf_conv(x$fitted_g, x$data)
+  n = length(cdfhat)
+  graphics::plot((1 : n) / (n + 1), sort(cdfhat), 
+                 pch = 19, cex = 0.25,
+                 xlim = c(0, 1), ylim = c(0, 1),
+                 xlab = "Uniform Theoretical Quantile", ylab = "Estimated Predictive Quantile",
+                 main = "Diagnostic Plot for ASH",
+                 ...)
+  abline(0, 1, lty = 2, col = "red")
+}
```
Comment (stephens999) [inline R/ashutility.R]: can you return(cdfhat) in case someone wants to get access to it?

File: R/ashutility.R [hunk 2]
```diff
@@ -57,6 +57,28 @@ plot.ash = function(x,...,xmin,xmax){
...
+#'
+#' @description Plot the cdf of the fitted predictive distribution at observations
+#'
+#' @param x the fitted ash object
+#' @param ... Arguments to be passed to methods,such as graphical parameters (see \code{\link[graphics]{plot}})
+#' @details None
+#'
+#' @export
+#'
+diag.ash = function(x, ...){
```
Comment (stephens999) [inline R/ashutility.R]: can we call it plot_diagnostic ?

**Comment by LSun** [comment]
removed copied code from plot_diagnostic, used my own simpler implementation
implemented uniform prior + normal likelihood
uniform prior + t likelihood not yet implemented, can be implemented in future

**Comment by stephens999** [comment]
just two things:
i) can you add plot.it=TRUE flag to the plot function? Then if it is false don't actually plot the plot? That would allow someone to use this function to compute the cdf values only if they want to.
ii) any particularly reason you return it as a list() with a single element, rather than just return a matrix? If not, can you just return a matrix?

**Comment by LSun** [comment]
i) sure it's done.
ii) i just want to give the output a name.  no particular reasons, just thought a name would be informative i guess?  now it's removed and the function just returns a matrix.

**Pull request merged**